misc/config_tools/schema/checks/cpu_assignment.xsd

```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+<xs:schema xml:id="root"
+	   xmlns:xs="http://www.w3.org/2001/XMLSchema"
+	   xmlns:acrn="https://projectacrn.org">
+
+  <xs:assert test="every $vm in /acrn-config/vm satisfies
+                   count(distinct-values($vm//cpu_affinity/pcpu_id)) = count($vm//cpu_affinity/pcpu_id)">
+    <xs:annotation acrn:severity="warning">
```

Comment:
It is possible to have an XML schema for an XML schema file. I can draft one to ensure correct annotation to the assertions.

Reply:
I just pushed another commit which adds XML schema files that checks the checker. The script `misc/config_tools/schema/checks/check_the_checkers.sh` will apply that schema to all the XSD files under `misc/config_tools/schema/checks` using `xmllint`.

Of course you need `xmllint` (which is part of `libxml2`) for this. I'm not using `xmlschema-validate` (which is provided by the `xmlschema` Python module) because that tool does not allow redefinition of XSD elements such as `xs:element`.